### PR TITLE
Feature/91 support for like operator

### LINKED ISSUE #91: Add suport for LIKE operator
Acceptance criteria:

I can build statements like SELECT * FROM BOOKS WHERE ISBN LIKE '123%'.

### PR DESCRIPTION
**Pull request by AnastasiiaSergienko**
Closes #91

### CODE REVIEW

File: doc/changes/changes_4.2.0.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+Code Name: LIKE predicate support
+
+## Features / Enhancements
+ 
+* #91: Added support for LIKE predicate.
+
+## Dependency updates
+
+* Added org.junit.jupiter:junit-jupiter:5.7.0
+* Updated org.mockito:mockito-core:jar:3.3.3 to version 3.5.13
```
Comment (jakobbraun) [inline doc/changes/changes_4.2.0.md:12]: wrong format.
In the RR template (https://github.com/exasol/release-robot/blob/master/doc/user_guide/changes_file_template.md) the following format is suggested:
`Updated <dependency groupId>:<dependency artifactId> from to `
I wrote a converter:
[converter.zip](https://github.com/exasol/sql-statement-builder/files/5364790/converter.zip)
Just place the two scripts in the folder above your projects root, and run from the project's root `../dependencyUpdate.sh`

File: pom.xml [hunk 2]
```diff
@@ -143,7 +128,7 @@
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-surefire-plugin</artifactId>
-                <version>${maven.surefire.version}</version>
+                <version>3.0.0-M4</version>
```
Comment (jakobbraun) [inline pom.xml:131]: Why did you remove the property?

**Review by jakobbraun — CHANGES_REQUESTED** [review]

File: doc/changes/changes_4.2.0.md (hunk 1 above)
Comment (AnastasiiaSergienko) [reply doc/changes/changes_4.2.0.md:12]: The format is not really checked somewhere and it's not stable yet, so we can change it to this one because it's easier to copy-paste after the updates

File: pom.xml (hunk 2 above)
Comment (AnastasiiaSergienko) [reply pom.xml:131]: Because we use the version in only one place, so it went directly to its dependency.

File: doc/changes/changes_4.2.0.md (hunk 1 above)
Comment (jakobbraun) [reply doc/changes/changes_4.2.0.md:12]: well but It is not readable. The `:jar` has no meaning to customers. Using the script I sent you can still copy & paste.

File: pom.xml (hunk 2 above)
Comment (jakobbraun) [reply pom.xml:131]: :+1:

Comment (jakobbraun) [reply pom.xml:131]: Maybe we should do this also if we use it in two places. I'm not sure if the versions plugin can update it otherwise.

**Review by jakobbraun — APPROVED** [review]

File: pom.xml (hunk 2 above)
Comment (AnastasiiaSergienko) [reply pom.xml:131]: It can, it has `update-properties`

File: doc/changes/changes_4.2.0.md (hunk 1 above)
Comment (AnastasiiaSergienko) [reply doc/changes/changes_4.2.0.md:12]: I also was not sure about `jar`, probably is can be removed. But the rest can be changed to this one, as it's already used in the plugin
